### PR TITLE
Caffe/Keras support dilation and deconvolutions + other minor fixes

### PR DESCRIPTION
**Pull request by gzuidhof**
I was looking to convert a [Caffe trained colorization network](http://richzhang.github.io/colorization/) to Keras, and a few things were missing, so I added them. I didn't succeed fully :(, so whether this all works or not is not entirely certain. It is possible that some filters need flipping or something, I am unable to test this at this moment (don't have Caffe configured on this machine).

Changes:
* **Support deconvolutions/transposed convolutions**, both in Caffe input and Keras output. 
* **Dilation factors in any type of convolution**, both in Caffe input and Keras output. 
* **Ignore silence layers in Caffe graphs**
* **Minor fix:** BatchNorm weights of shape `(1,1,1,64)`or similar, are now reshaped to `(64,)` to prevent broadcasting issues.
* **Fix typo:** *paramters -> parameters*.

There's also a stub for support for Scale layer. I figured a scale layer is equivalent to a BatchNorm layer with gamma weights equal to the scale factor, no bias, var=1 and mean=0. Unfortunately the Scale layer doesn't necessarily have learned weights, so it doesn't appear in the graph lists. I'm not sure where and how to add it. Maybe one of the main contributors can have a look at this and finish this up? :)

### CODE REVIEW

**Comment by kitstar** [comment]
Thanks for your contribution!

In our implementation, the scale layer with BN layer is fused to generate all the related parameter, which has been handled already. You can check the fused related code. Thanks.

**Comment by gzuidhof** [comment]
Scale layers without BN is not handled yet

**Comment by kitstar** [comment]
Hi @gzuidhof , we will fill the scale standalone related code later. Thanks.

**Comment by kitstar** [comment]
Hi @gzuidhof   just fixed the lack of 'dilation' parameter of pooling layer. ref this [commit](https://github.com/Microsoft/MMdnn/commit/ae495db3706a1c62e2f40f311d35e7e598a10dab).
